worker/uploader.go

```diff
@@ -134,16 +138,49 @@ outer:
 				err: err,
 			}:
 			}
+		}
+	}
+}
 
-			// track the error, ignore gracefully closed streams and canceled overdrives
-			canceledOverdrive := req.done() && req.overdrive && err != nil
-			if !canceledOverdrive && !isClosedStream(err) {
-				u.trackSectorUpload(err, elapsed)
-			} else {
-				u.logger.Debugw("not tracking sector upload metric", zap.Error(err))
-			}
+func handleSectorUpload(uploadErr error, uploadDuration, totalDuration time.Duration, overdrive bool, logger *zap.SugaredLogger) (success bool, failure bool, uploadEstimateMS float64, uploadSpeedBytesPerMS float64) {
+	// special case, uploader will refresh and the request will be requeued
+	if errors.Is(uploadErr, errMaxRevisionReached) {
+		logger.Debugw("sector upload failure was ignored", "uploadError", uploadErr, "uploadDuration", uploadDuration, "totalDuration", totalDuration, "overdrive", overdrive)
+		return
```

Comment:
It's fine to keep the named return vars as descriptions but I'd still explicitly return values here for readability.